app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt

```diff
@@ -189,6 +200,24 @@ class DownloadFragment : Fragment() {
         view?.setLayoutWidth(bytes)
     }
 
+    private fun openLocalVideo() {
+        val intent = Intent()
+            .setAction(Intent.ACTION_GET_CONTENT)
+            .setType("video/*")
+            .addCategory(Intent.CATEGORY_OPENABLE)
+            .addFlags(FLAG_GRANT_READ_URI_PERMISSION) // Request temporary access
+        try {
+            videoResultLauncher.launch(
+                Intent.createChooser(
+                    intent,
+                    getString(R.string.open_local_video)
+                )
+            )
+        } catch (t: ActivityNotFoundException) {
+            t.printStackTrace()
```

Comment:
logError ?

Reply:
in most cases the only exception will be the Files app not found in some shitty roms, so now it is in normal safe call to catch and log any error